groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1431,10708 @@ void Optional_DataImp<TYPE>::emplace(
+        // memory.  'rhs' is left in a valid but unspecified state.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        this->emplace(BSLS_COMPILERFEATURES_FORWARD(ANY_TYPE, rhs));
+    }
+
+    template <class ANY_TYPE>
+    explicit optional(bsl::allocator_arg_t,
+                      allocator_type                              allocator,
+                      BSLS_COMPILERFEATURES_FORWARD_REF(ANY_TYPE) rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_ANYTYPE)
+    : d_allocator(allocator)
+        // Create an 'optional' object having the same value as the specified
+        // 'rhs' object by forwarding the contents of 'rhs' to the
+        // newly-created object.  Use the specified 'allocator' to supply
+        // memory.  'rhs' is left in a valid but unspecified state.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        this->emplace(BSLS_COMPILERFEATURES_FORWARD(ANY_TYPE, rhs));
+    }
+
+    template <class ANY_TYPE>
+    explicit optional(bsl::allocator_arg_t,
+                      allocator_type            allocator,
+                      const optional<ANY_TYPE>& rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_OPTIONAL_LVAL)
+    : d_allocator(allocator)
+        // If specified 'rhs' contains a value, initialize the 'value_type'
+        // object with '*rhs'.  Otherwise, create a disengaged 'optional'.  Use
+        // the specified 'allocator' to supply memory.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        if (rhs.has_value()) {
+            this->emplace(rhs.value());
+        }
+    }
+
+#ifdef BSLS_COMPILERFEATURES_SUPPORT_RVALUE_REFERENCES
+    // 'MovableRef' prevents correct type deduction in C++11 when used with
+    // 'optional<ANY_TYPE>'.  This constructor needs to be defined in terms of
+    // rvalue reference in C++11.  In C++03, this type deduction issue does not
+    // exist due to the nature of C++03 MovableRef implementation and usage.
+    // Consequently, a 'MovableRef' equivalent constructor needs to be provided
+    // in C++03 (see below).
+    template <class ANY_TYPE>
+    explicit optional(bsl::allocator_arg_t,
+                      allocator_type       allocator,
+                      optional<ANY_TYPE>&& rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_OPTIONAL_RVAL)
+    : d_allocator(allocator)
+        // If specified 'rhs' contains a value, initialize the 'value_type'
+        // object by move construction from '*rhs'.  Otherwise, create a
+        // disengaged 'optional'.  Use the specified 'allocator' to supply
+        // memory.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        optional<ANY_TYPE>& lvalue = rhs;
```

Comment:
No need for lvalue here.

Reply:
fixed as part of the ' use std instead of MoveUtil in C++11 land' comment.